openshift/osd-network-verifier
# Firewall Config Script Updated

PR content:
a go script to create a VPC and firewall for OSD-network-verifier testing.
How to test:

if you want to pass credentials via the ENV:
  + osdctl account cli -i $account_id -p osd-staging-2 -o env
  + run 'go build firewallConfig.go'
  + AWS_ACCESS_KEY_ID= AWS_SECRET_ACCESS_KEY= AWS_SESSION_TOKEN= REGION= ./firewallConfig

if you want to pass in a profile :
  + run 'go build firewallConfig.go'
  + AWS_PROFILE=$profile AWS_REGION=$region ./firewallConfig

if you want to pass in a profile via  flag:
  + run 'go build firewallConfig.go'
  +./firewallConfig -p $your-profile -r $region

How to delete the firewall and the resources created by the script:

Remove all routes in all the routes tables
Delete NAT Gateway
Delete the Firewall
After the Firewall is done deleting, delete the subnets (3 subnets)
Detach the IGW from the VPC and delete the IGW
Delete the route tables (the one that’s marked main will deleted with the vpc)-> 4 route tables to delete
Delete the VPC
Delete the firewall policy
Delete the rule group
open your AWS console and see that VPC and Firewall have been created.

Related Jira ticket: https://issues.redhat.com/browse/OSD-10310
